// mediasoup/node/lib/fbs/worker/close-notification.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloseNotificationT = exports.CloseNotification = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, @typescript-eslint/no-non-null-assertion */
const flatbuffers = require("flatbuffers");
class CloseNotification {
    bb = null;
    bb_pos = 0;
    __init(i, bb) {
        this.bb_pos = i;
        this.bb = bb;
        return this;
    }
    static getRootAsCloseNotification(bb, obj) {
        return (obj || new CloseNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static getSizePrefixedRootAsCloseNotification(bb, obj) {
        bb.setPosition(bb.position() + flatbuffers.SIZE_PREFIX_LENGTH);
        return (obj || new CloseNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static startCloseNotification(builder) {
        builder.startObject(0);
    }
    static endCloseNotification(builder) {
        const offset = builder.endObject();
        return offset;
    }
    static createCloseNotification(builder) {
        CloseNotification.startCloseNotification(builder);
        return CloseNotification.endCloseNotification(builder);
    }
    unpack() {
        return new CloseNotificationT();
    }
    unpackTo(_o) { }
}
exports.CloseNotification = CloseNotification;
class CloseNotificationT {
    constructor() { }
    pack(builder) {
        return CloseNotification.createCloseNotification(builder);
    }
}
exports.CloseNotificationT = CloseNotificationT;
