// mediasoup/node/lib/fbs/worker/close-worker-notification.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloseWorkerNotificationT = exports.CloseWorkerNotification = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, @typescript-eslint/no-non-null-assertion */
const flatbuffers = require("flatbuffers");
class CloseWorkerNotification {
    bb = null;
    bb_pos = 0;
    __init(i, bb) {
        this.bb_pos = i;
        this.bb = bb;
        return this;
    }
    static getRootAsCloseWorkerNotification(bb, obj) {
        return (obj || new CloseWorkerNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static getSizePrefixedRootAsCloseWorkerNotification(bb, obj) {
        bb.setPosition(bb.position() + flatbuffers.SIZE_PREFIX_LENGTH);
        return (obj || new CloseWorkerNotification()).__init(bb.readInt32(bb.position()) + bb.position(), bb);
    }
    static startCloseWorkerNotification(builder) {
        builder.startObject(0);
    }
    static endCloseWorkerNotification(builder) {
        const offset = builder.endObject();
        return offset;
    }
    static createCloseWorkerNotification(builder) {
        CloseWorkerNotification.startCloseWorkerNotification(builder);
        return CloseWorkerNotification.endCloseWorkerNotification(builder);
    }
    unpack() {
        return new CloseWorkerNotificationT();
    }
    unpackTo(_o) { }
}
exports.CloseWorkerNotification = CloseWorkerNotification;
class CloseWorkerNotificationT {
    constructor() { }
    pack(builder) {
        return CloseWorkerNotification.createCloseWorkerNotification(builder);
    }
}
exports.CloseWorkerNotificationT = CloseWorkerNotificationT;
