// mediasoup/node/lib/fbs/request/method.js
"use strict";
// automatically generated by the FlatBuffers compiler, do not modify
Object.defineProperty(exports, "__esModule", { value: true });
exports.Method = void 0;
/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-explicit-any, @typescript-eslint/no-non-null-assertion */
var Method;
(function (Method) {
    Method[Method["WORKER_DUMP"] = 0] = "WORKER_DUMP";
    Method[Method["WORKER_GET_RESOURCE_USAGE"] = 1] = "WORKER_GET_RESOURCE_USAGE";
    Method[Method["WORKER_UPDATE_SETTINGS"] = 2] = "WORKER_UPDATE_SETTINGS";
    Method[Method["WORKER_CREATE_WEBRTCSERVER"] = 3] = "WORKER_CREATE_WEBRTCSERVER";
    Method[Method["WORKER_CREATE_ROUTER"] = 4] = "WORKER_CREATE_ROUTER";
    Method[Method["WORKER_WEBRTCSERVER_CLOSE"] = 5] = "WORKER_WEBRTCSERVER_CLOSE";
    Method[Method["WORKER_CLOSE_ROUTER"] = 6] = "WORKER_CLOSE_ROUTER";
    Method[Method["WEBRTCSERVER_DUMP"] = 7] = "WEBRTCSERVER_DUMP";
    Method[Method["ROUTER_DUMP"] = 8] = "ROUTER_DUMP";
    Method[Method["ROUTER_CREATE_WEBRTCTRANSPORT"] = 9] = "ROUTER_CREATE_WEBRTCTRANSPORT";
    Method[Method["ROUTER_CREATE_WEBRTCTRANSPORT_WITH_SERVER"] = 10] = "ROUTER_CREATE_WEBRTCTRANSPORT_WITH_SERVER";
    Method[Method["ROUTER_CREATE_PLAINTRANSPORT"] = 11] = "ROUTER_CREATE_PLAINTRANSPORT";
    Method[Method["ROUTER_CREATE_PIPETRANSPORT"] = 12] = "ROUTER_CREATE_PIPETRANSPORT";
    Method[Method["ROUTER_CREATE_DIRECTTRANSPORT"] = 13] = "ROUTER_CREATE_DIRECTTRANSPORT";
    Method[Method["ROUTER_CLOSE_TRANSPORT"] = 14] = "ROUTER_CLOSE_TRANSPORT";
    Method[Method["ROUTER_CREATE_ACTIVESPEAKEROBSERVER"] = 15] = "ROUTER_CREATE_ACTIVESPEAKEROBSERVER";
    Method[Method["ROUTER_CREATE_AUDIOLEVELOBSERVER"] = 16] = "ROUTER_CREATE_AUDIOLEVELOBSERVER";
    Method[Method["ROUTER_CLOSE_RTPOBSERVER"] = 17] = "ROUTER_CLOSE_RTPOBSERVER";
    Method[Method["TRANSPORT_DUMP"] = 18] = "TRANSPORT_DUMP";
    Method[Method["TRANSPORT_GET_STATS"] = 19] = "TRANSPORT_GET_STATS";
    Method[Method["TRANSPORT_CONNECT"] = 20] = "TRANSPORT_CONNECT";
    Method[Method["TRANSPORT_SET_MAX_INCOMING_BITRATE"] = 21] = "TRANSPORT_SET_MAX_INCOMING_BITRATE";
    Method[Method["TRANSPORT_SET_MAX_OUTGOING_BITRATE"] = 22] = "TRANSPORT_SET_MAX_OUTGOING_BITRATE";
    Method[Method["TRANSPORT_SET_MIN_OUTGOING_BITRATE"] = 23] = "TRANSPORT_SET_MIN_OUTGOING_BITRATE";
    Method[Method["TRANSPORT_RESTART_ICE"] = 24] = "TRANSPORT_RESTART_ICE";
    Method[Method["TRANSPORT_PRODUCE"] = 25] = "TRANSPORT_PRODUCE";
    Method[Method["TRANSPORT_PRODUCE_DATA"] = 26] = "TRANSPORT_PRODUCE_DATA";
    Method[Method["TRANSPORT_CONSUME"] = 27] = "TRANSPORT_CONSUME";
    Method[Method["TRANSPORT_CONSUME_DATA"] = 28] = "TRANSPORT_CONSUME_DATA";
    Method[Method["TRANSPORT_ENABLE_TRACE_EVENT"] = 29] = "TRANSPORT_ENABLE_TRACE_EVENT";
    Method[Method["TRANSPORT_CLOSE_PRODUCER"] = 30] = "TRANSPORT_CLOSE_PRODUCER";
    Method[Method["TRANSPORT_CLOSE_CONSUMER"] = 31] = "TRANSPORT_CLOSE_CONSUMER";
    Method[Method["TRANSPORT_CLOSE_DATAPRODUCER"] = 32] = "TRANSPORT_CLOSE_DATAPRODUCER";
    Method[Method["TRANSPORT_CLOSE_DATACONSUMER"] = 33] = "TRANSPORT_CLOSE_DATACONSUMER";
    Method[Method["PLAINTRANSPORT_CONNECT"] = 34] = "PLAINTRANSPORT_CONNECT";
    Method[Method["PIPETRANSPORT_CONNECT"] = 35] = "PIPETRANSPORT_CONNECT";
    Method[Method["WEBRTCTRANSPORT_CONNECT"] = 36] = "WEBRTCTRANSPORT_CONNECT";
    Method[Method["PRODUCER_DUMP"] = 37] = "PRODUCER_DUMP";
    Method[Method["PRODUCER_GET_STATS"] = 38] = "PRODUCER_GET_STATS";
    Method[Method["PRODUCER_PAUSE"] = 39] = "PRODUCER_PAUSE";
    Method[Method["PRODUCER_RESUME"] = 40] = "PRODUCER_RESUME";
    Method[Method["PRODUCER_ENABLE_TRACE_EVENT"] = 41] = "PRODUCER_ENABLE_TRACE_EVENT";
    Method[Method["CONSUMER_DUMP"] = 42] = "CONSUMER_DUMP";
    Method[Method["CONSUMER_GET_STATS"] = 43] = "CONSUMER_GET_STATS";
    Method[Method["CONSUMER_PAUSE"] = 44] = "CONSUMER_PAUSE";
    Method[Method["CONSUMER_RESUME"] = 45] = "CONSUMER_RESUME";
    Method[Method["CONSUMER_SET_PREFERRED_LAYERS"] = 46] = "CONSUMER_SET_PREFERRED_LAYERS";
    Method[Method["CONSUMER_SET_PRIORITY"] = 47] = "CONSUMER_SET_PRIORITY";
    Method[Method["CONSUMER_REQUEST_KEY_FRAME"] = 48] = "CONSUMER_REQUEST_KEY_FRAME";
    Method[Method["CONSUMER_ENABLE_TRACE_EVENT"] = 49] = "CONSUMER_ENABLE_TRACE_EVENT";
    Method[Method["DATAPRODUCER_DUMP"] = 50] = "DATAPRODUCER_DUMP";
    Method[Method["DATAPRODUCER_GET_STATS"] = 51] = "DATAPRODUCER_GET_STATS";
    Method[Method["DATAPRODUCER_PAUSE"] = 52] = "DATAPRODUCER_PAUSE";
    Method[Method["DATAPRODUCER_RESUME"] = 53] = "DATAPRODUCER_RESUME";
    Method[Method["DATACONSUMER_DUMP"] = 54] = "DATACONSUMER_DUMP";
    Method[Method["DATACONSUMER_GET_STATS"] = 55] = "DATACONSUMER_GET_STATS";
    Method[Method["DATACONSUMER_PAUSE"] = 56] = "DATACONSUMER_PAUSE";
    Method[Method["DATACONSUMER_RESUME"] = 57] = "DATACONSUMER_RESUME";
    Method[Method["DATACONSUMER_GET_BUFFERED_AMOUNT"] = 58] = "DATACONSUMER_GET_BUFFERED_AMOUNT";
    Method[Method["DATACONSUMER_SET_BUFFERED_AMOUNT_LOW_THRESHOLD"] = 59] = "DATACONSUMER_SET_BUFFERED_AMOUNT_LOW_THRESHOLD";
    Method[Method["DATACONSUMER_SEND"] = 60] = "DATACONSUMER_SEND";
    Method[Method["DATACONSUMER_SET_SUBCHANNELS"] = 61] = "DATACONSUMER_SET_SUBCHANNELS";
    Method[Method["DATACONSUMER_ADD_SUBCHANNEL"] = 62] = "DATACONSUMER_ADD_SUBCHANNEL";
    Method[Method["DATACONSUMER_REMOVE_SUBCHANNEL"] = 63] = "DATACONSUMER_REMOVE_SUBCHANNEL";
    Method[Method["RTPOBSERVER_PAUSE"] = 64] = "RTPOBSERVER_PAUSE";
    Method[Method["RTPOBSERVER_RESUME"] = 65] = "RTPOBSERVER_RESUME";
    Method[Method["RTPOBSERVER_ADD_PRODUCER"] = 66] = "RTPOBSERVER_ADD_PRODUCER";
    Method[Method["RTPOBSERVER_REMOVE_PRODUCER"] = 67] = "RTPOBSERVER_REMOVE_PRODUCER";
})(Method || (exports.Method = Method = {}));
